guides/development-workflow.md

```diff
@@ -0,0 +1,284 @@
+  3. merge the pull request
+  4. manually merge the pull request back into develop
+  ```
+  git checkout develop
+  git pull
+  git merge origin/master
+  git push
+  ```
+  5. move the card into 'Deployed to production'
+
+---
+
+## Branching
+
+#### How to make a branch
+
+Create the branch off of develop:
+```
+git fetch
+git checkout develop
+git branch <branch-name>
+git checkout <branch-name>
+```
+
+**Naming the branch**
+
+Having a convention allows for:
+
+* all branches connected to a story or ticket are easily connected now and in the future
+* others can easily identify the intent of the contents
+
+**Convention:**
+
+* Feature - A branch that adds a new feature, as defined by the specified story
+* Fix - A branch that corrects a problem in a feature already merged into develop
+* Chore - Chore branches are used for routine tasks or tickets which are not emergencies
+* Hotfix - A branch that adds an urgent fix to a problem that affects production. These branches are based on master and do not go via develop, so must be also be merged into develop when deployed
+
+```
+feature/{STORY ID}-{CONCISE TITLE}
+fix/{STORY/TICKET ID}-{CONCISE TITLE}
+chore/{STORY ID}-{CONCISE TITLE}
+hotfix/{TICKET ID}-{CONCISE TITLE}
+```
+
+**Examples:**
+```
+feature/523797477-add-logging-to-registration
+fix/423797477-logging-happens-in-both-environments
+chore/1934729239-reduce-caching-for-contact-details
+hotfix/253947623-remove-breaking-change-to-repair-creation
+```
+
+#### Update your branch with the latest changes on develop (rebasing)
+
+When working in a team other pull requests will often get merged in before yours is finished  It's important to make sure you're developing against the latest version of the code base as much as possible.
+
+You can do this by rebasing your work on top of the develop branch.
+
+```
+git checkout <branch-name>
+git rebase origin/develop
+git push --force-with-lease
+```
+
+During the rebase there may be conflicts which should be resolved at this point. If in doubt talk to the other developer who you're conflicting with and reason out what the correct state should be.
+
+Forcing pushing in this situation is the only time it should be necessary, adding `--force-with-lease` ensures you don't accidentally overwrite someone else's additions.
+
+## Committing
+
+Writing a good commit message serves three important purposes.
+
+* to speed up the review process
+* to encourage individual commits to be well small, bounded and concise
+* to help future maintainers (it could be you) in say 3 years time, to find out why a particular change was made and what context was relevant at the time
+
+**Convention:**
+```
+(feature): when we are adding something new
+(fix): when we are fixing something existing
+(refactor): when we are changing the implementation of something existing without changing its behaviour
+(chore): catch all for when none of the above apply and there is no immediate user need eg. gem/package updates
```

Comment:
I had the same reservations when we tried this but have since seen the benefit in them. Plenty of other teams use this too like [Angluar](https://github.com/angular/angular).

For me it gives us: 

Communication: it removes any ambiguity around the commit title.

If I'm looking at a commit that has `(refactor):` then I don't have to interpret the title as to what the code should be doing. I know regardless of what the title says that the code should not add or change any behaviour, just implementation. I should be able to revert this commit without impacting any functionanlity or breaking any tests.

Before using a convention it was easy to have large commits that solved a problem, they could easily become behemoths because there was no decision point. This harmful to the project and our velocity because they were doing far too much at once: adding new things, fixing others in passing and refactoring. This was impossible for someone else to read and give me feedback on, it didn't tell a clear story. 

Now however when reviewing a pull request I will see:

- refactor: foo
- refactor: bar
- fix: xyz
- refactor: baz
- feature: zoo

I can see in seconds that the author has taken some refactoring steps before adding in their feature (👍) at the end, making a fix a long the way. I understand what to focus on and what can be removed if anything becomes contentious, chopping it speeds up the time it will take to merge.